imagebuildah/util.go

```diff
@@ -99,6 +120,17 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 		}
 		return name, subdir, nil
 	}
+	if url == "-" {
+		err = stdinToDirectory(name)
+		if err != nil {
+			if err2 := os.Remove(name); err2 != nil {
+				logrus.Debugf("error removing temporary directory %q: %v", name, err2)
```

Comment:
Shouldn't this be removing temporary file?

Reply:
Or should this be os.RemoveAll(name)

Reply:
I haven't looked too much into how the unpacking works as used here as this is based on the above `downloadToDirectory` function above.

If we decide to change `os.Remove` to `os.RemoveAll`, we should probably update `downloadToDirectory` in a parent commit first so the behavior of loading context via stdin or url remain consistent.

Now looking into the unpacking, as far as I can tell at this time, this is what occurs in both `stdinToDirectory` and `downloadToDirectory.

1. If untar dies because input is not a tar, it is assumed to be a Dockerfile, and the output is written to `tmpdir/Dockerfile` without any further checking, this is the expected path.

2. If untar dies because of an error with the tar, such as corruption, short-read, etc, then the files are left behind and the tar is written to Dockerfile, this stage continues as a success, and I presume the folder gets cleaned up in the calling function after the build fails because Dockerfile is actually a tar.

3. If the tar unpacking dies because it runs out of diskspace, we attempt to create the Dockerfile and this will fail due to diskspace, then `os.Remove` will fail, and we will leave behind a full filesystem on exit.

4. It's not a tar, so no tar files are created, but the filesystem doesn't have enough space for the Dockerfile, so we leave behind a half written Dockerfile, then fail to cleanup because the directory is not empty.

As far as I can tell, the only time these functions are going to truly fail is in scenario 3 and 4.  2 is probably somewhat undefined behavior that's not great.

---

Before I switch this to `os.RemoveAll`, I don't see anything that would make this dangerous, but is there anything I'm not seeing here?  I'd hate to cause someone's files to go poof unexpectedly.

---

If we used something like `libmagic` we could figure out if the input was compressed/tar before deciding what to do with it, which would improve our ability to error-handle here, and means I could use `bufio` to not read the whole tar into memory, and use `Peek` to figure out if we have a tar or a compressed stream which we would assume to be a tar.  But that leaves two questions.  What's the policy on CGO (I don't see any pure-go implementations for this)?  And would moving this much logic here be acceptable?

Reply:
One issue with CGO would be this tool being vendored into other tools like Podman and OpenShift builder, and perhaps others (ansible?)  

The problem you are worried about is the user specifying a NAME path that containers other files correct?  So that RemoveAll would somehow remove files that the user did not download.  Could we check before the untar if the NAME path exists and cause an error?  Is the NAME Path supposed to be a randomly created file in a random directory then we would not need to worry about this?

Reply:
`name` is created by `ioutil.TempDir(dir, prefix)`
So we can `removeall` as long as `ioutil.TempDir` always gives us a temporary directory.